q2_diversity_lib/_alpha/__init__.py

```diff
@@ -6,5 +6,7 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
-from ._method import (faith_pd)
-__all__ = ['faith_pd']
+from ._method import (faith_pd, observed_features, pielou_evenness,
```

Comment:
Putting this comment here because no-where else to pin to --- I think this plugin should expose the alpha (and someday beta) subpackages, kind of like what we do in [q2-types](https://github.com/qiime2/q2-types/tree/master/q2_types). The `_alpha` syntax, while only a convention, does tend to signal "this is private code" --- I think it should be named `alpha`, and the methods should be part of the subpackage's public API.

Reply:
I like!

Reply:
> If I do so, what is the best way to de-clutter the namespace encountered by Artifact API users?

The same approach should be used for _all_ solutions: exporting an `__all__`. Also, this doesn't impact the Artifact API at all, only the native python API (that is, your example above is skipping the QIIME 2 framework and just importing functionality directly using native python).

I agree with @ebolyen's suggestion to drop the subpackages --- they only matter for the structure of the source code, modules and subpackages both provide the same "feel" for python users. Subpackages are just modules with more files (sub-modules). I was just assuming you had plans for more files within your subpackage.

I would go with the `alpha.py` and `beta.py` modules, and make sure to define and export an `__all__` in the root package's `__init__.py`.

Reply:
Please see the following for more details RE `__all__`: https://docs.python.org/3/tutorial/modules.html#importing-from-a-package

Reply:
Thanks for the guidance, @thermokarst, especially re: artifact vs standard Python API. I read the modules tutorial this morning and understand what's going on there - just missing the forest for the trees. I'll carry on with this, and can always break things out into subpackages if that feels valuable later.